File: app/src/main/java/com/snaptiongame/app/data/converters/GameActionConverter.java
package com.snaptiongame.app.data.converters;

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;
import com.snaptiongame.app.data.models.GameAction;

import java.lang.reflect.Type;

/**
 * @author Tyler Wong
 */

public class GameActionConverter implements JsonSerializer<GameAction> {

    @Override
    public JsonElement serialize(GameAction src, Type typeOfSrc, JsonSerializationContext context) {
        JsonObject json = new JsonObject();
        json.addProperty(src.getType(), src.getTargetId());
        json.addProperty(src.getChoiceType(), src.getChoice());
        return json;
    }
}

File: app/src/main/java/com/snaptiongame/app/data/converters/GameActionConverter.kt
package com.snaptiongame.app.data.converters

import com.google.gson.JsonElement
import com.google.gson.JsonObject
import com.google.gson.JsonSerializationContext
import com.google.gson.JsonSerializer
import com.snaptiongame.app.data.models.GameAction

import java.lang.reflect.Type

/**
 * @author Tyler Wong
 */

class GameActionConverter : JsonSerializer<GameAction> {

    override fun serialize(src: GameAction, typeOfSrc: Type, context: JsonSerializationContext): JsonElement {
        val json = JsonObject()
        json.addProperty(src.type, src.targetId)
        json.addProperty(src.choiceType, src.choice)
        return json
    }
}
